presto-main/src/main/java/com/facebook/presto/sql/FunctionInvoker.java

```diff
@@ -53,37 +52,53 @@ public Object invoke(Signature function, ConnectorSession session, List<Object>
         ScalarFunctionImplementation implementation = registry.getScalarFunctionImplementation(function);
         MethodHandle method = implementation.getMethodHandle();
 
+        // handle function on instance method, to allow use of fields
+        method = bindInstanceFactory(method, implementation);
+
         if (method.type().parameterCount() > 0 && method.type().parameterType(0) == ConnectorSession.class) {
             method = method.bindTo(session);
         }
-        Iterator<Object> iterator = arguments.iterator();
         List<Object> actualArguments = new ArrayList<>(arguments.size() + 1);
-        for (int i = 0; i < method.type().parameterCount(); i++) {
-            Class<?> parameterType = method.type().parameterType(i);
-            checkArgument(iterator.hasNext(), "Not enough arguments provided for method: %s", method.type());
-            Object argument = iterator.next();
+        for (int i = 0; i < arguments.size(); i++) {
+            Object argument = arguments.get(i);
+            if (implementation.getLambdaInterface().get(i).isPresent() && !MethodHandle.class.equals(implementation.getLambdaInterface().get(i).get())) {
```

Comment:
I think extraction of `implementation.getLambdaInterface().get(i)` can improve readability.